omod/src/main/webapp/resources/scripts/emr.js

```diff
@@ -1,389 +1,405 @@
+        var ret = "?";
+        if (options) {
+            for (key in options) {
+                var val = options[key];
+                if ($.isArray(val)) {
+                    for (ind in val) {
+                        ret += key + '=' + encodeURIComponent(val[ind]) + '&';
+                    }
+                } else {
+                    ret += key + '=' + encodeURIComponent(val) + '&';
+                }
+            }
+        }
+        return ret;
+    };
+
+    var requireOptions = function() {
+        var opts = arguments[0];
+        for (var i = 1; i < arguments.length; ++i) {
+            if (!opts[arguments[i]]) {
+                throw "Missing option: " + arguments[i];
+            }
+        }
+    };
+
+    var jqObject = $();
+
+    return {
+
+        // just used in testing so we can replace the instance with a mock or spy as needed
+        setJqObject: function(jqueryInstanceToSet)  {
+            jqObject = jqueryInstanceToSet;
+        },
+
+        loadMessages: function(codes, callback) {
+            if (!window.messages) {
+                window.messages = {};
+            }
+            emr.getFragmentActionWithCallback("uicommons", "messages", "get", { codes: codes }, function(response) {
+                for (var code in response) {
+                    window.messages[code] = response[code];
+                }
+                if (callback) {
+                    callback(window.messages);
+                }
+            });
+        },
+
+        loadGlobalProperties: function(properties, callback) {
+            if (!window.globalProperties) {
+                window.globalProperties = {};
+            }
+            emr.getFragmentActionWithCallback("uicommons", "globalProperties", "get", {properties: properties}, function(response) {
+                for (var property in response) {
+                    window.globalProperties[property] = response[property];
+                }
+                if (callback) {
+                    callback(window.globalProperties);
+                }
+            });
+        },
+
+        message: function(code, defaultText) {
+            if (window.messages) {
+                var translated = window.messages[code];
+                return translated ? translated : (defaultText ? defaultText : code);
+            } else {
+                return defaultText ? defaultText : code;
+            }
+        },
+
+        serverErrorMessage: function(response) {
+        	var errorMessgae = response.data.error.message;
+        	if (response.data.error.globalErrors != null) {
+            	response.data.error.globalErrors.forEach(function (val, idx){
+            		errorMessgae += " \n" + val.message;
+                }); 
+        	}
+        	if (response.data.error.fieldErrors != null 
+        		&& response.data.error.fieldErrors.name != null) {
+            	response.data.error.fieldErrors.name.forEach(function (val, idx){

```

Comment:
name was just an example field in that example json on the ticket, the field errors are map entries where the field's name is the key and the value is a list of errors for that field.